database/src/desktop/view/view.cc

```diff
@@ -152,7 +152,7 @@ std::vector<Event> View::ApplyOperation(
     const Variant* opt_complete_server_cache,
     std::vector<Change>* out_changes) {
   if (operation.type == Operation::kTypeMerge &&
-      !operation.source.query_params.has_value()) {
+      operation.source.query_params.has_value()) {
```

Comment:
Could you explain this change a bit?  Seems like a pretty fundamental turn.

Reply:
When I originally transcribed this code from the Java reference implementation I just got the condition exactly backwards. Whoops

The [reference implementation](https://github.com/firebase/firebase-android-sdk/blob/d92096257b3e750c72757fc7c4bc9d7393db2046/firebase-database/src/main/java/com/google/firebase/database/core/view/View.java#L163) had:

    if (operation.getType() == Operation.OperationType.Merge
        && operation.getSource().getQueryParams() != null) {

And checking for `!= null` is equivalent to checking if an optional's `.has_value()` is true.

Reply:
Gotcha, thanks.  👍